pkg/apis/serving/v1beta1/configuration_validation.go

```diff
@@ -61,3 +64,26 @@ func (cs *ConfigurationStatus) Validate(ctx context.Context) *apis.FieldError {
 func (csf *ConfigurationStatusFields) Validate(ctx context.Context) *apis.FieldError {
 	return nil
 }
+
+// ValidateLabels function validates service labels
+func (c *Configuration) ValidateLabels() *apis.FieldError {
+	var errs *apis.FieldError
+LabelLoop:
+	for key, val := range c.GetLabels() {
+		switch {
+		case key == config.VisibilityLabelKey:
+			errs = errs.Also(validateClusterVisbilityLabel(val))
+		case key == serving.RouteLabelKey:
+		case key == serving.ServiceLabelKey:
+			for _, ref := range c.GetOwnerReferences() {
+				if ref.Kind == "Service" && val == ref.Name {
```

Comment:
1. I think you want to check all, so probably not bail out early
2. if you invert the check, then continue would not be required in any case and label can go.

Reply:
good point. I did not consider the case of service owner references.